daemon/mgr/network.go

```diff
@@ -149,26 +150,86 @@ func (nm *NetworkManager) List(ctx context.Context, labels map[string]string) ([
 	return net, nil
 }
 
-// Get returns the information of network that specified name/id.
-func (nm *NetworkManager) Get(ctx context.Context, name string) (*types.Network, error) {
+// GetNetworkByName returns the information of network that specified name.
+func (nm *NetworkManager) GetNetworkByName(name string) (*types.Network, error) {
 	n, err := nm.controller.NetworkByName(name)
-	if err != nil {
-		if err == libnetwork.ErrNoSuchNetwork(name) {
-			return nil, errors.Wrap(errtypes.ErrNotfound, err.Error())
-		}
-		return nil, err
-	}
-
-	if n != nil {
+	if err == nil {
 		return &types.Network{
-			Name:    name,
+			Name:    n.Name(),
 			ID:      n.ID(),
 			Type:    n.Type(),
 			Network: n,
 		}, nil
 	}
+	return nil, err
+}
 
-	return nil, nil
+// GetNetworksByID returns a list of networks that specified id.
+func (nm *NetworkManager) GetNetworksByID(id string) []*types.Network {
+	var matchedNetworks []*types.Network
+
+	walker := func(nw libnetwork.Network) bool {
+		if strings.HasPrefix(nw.ID(), id) {
+			matchedNetwork := &types.Network{
+				Name:    nw.Name(),
+				ID:      nw.ID(),
+				Type:    nw.Type(),
+				Network: nw,
+			}
+			matchedNetworks = append(matchedNetworks, matchedNetwork)
+		}
+		return false
+	}
+	nm.controller.WalkNetworks(walker)
+	return matchedNetworks
+}
+
+// GetNetworkByID returns the information of network that specified id.
+func (nm *NetworkManager) GetNetworkByID(id string) (*types.Network, error) {
+	network, err := nm.controller.NetworkByID(id)
+	if err == nil {
+		return &types.Network{
+			Name:    network.Name(),
+			ID:      network.ID(),
+			Type:    network.Type(),
+			Network: network,
+		}, nil
+	}
+	if !isNoSuchNetworkError(err) {
+		return nil, err
+	}
+	matchedNetworks := nm.GetNetworksByID(id)
+	if len(matchedNetworks) == 0 {
+		return nil, errors.Wrap(errtypes.ErrNotfound, libnetwork.ErrNoSuchNetwork(id).Error())
```

Comment:
I do not think we should wrap this, since we have already indentify this is a `libnetwork.ErrNoSuchNetwork(id)`.
Or remove the latter one to keep the `errors.Wrap`?
WDYT? @faycheng